### PR TITLE
Completed task 7. POST /api/articles/:article_id/comments

### PR DESCRIPTION
**Pull request by juliaozm**
Submitting solution for task 7. POST /api/articles/:article_id/comments

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -40,4 +40,207 @@ describe('news-project', () => {
...
+            return request(app)
+            .post('/api/articles/1/comments')
+            .send(newComment)
+            .expect(404)
+            .then(({body: {message}}) => {
+                expect(message).toBe('This user does not exist');
+            })
+        })
+
+        test('POST: 400 - returns a message "Bad request" when a received object is empty', () => {
```
Comment (Hard-Luck) [inline __tests__/app.test.js]: I would change the wording of this to, request body

File: __tests__/app.test.js [hunk 2]
```diff
@@ -40,4 +40,207 @@ describe('news-project', () => {
...
+            }
+            return request(app)
+            .post('/api/articles/1/comments')
+            .send(newComment)
+            .expect(400)
+            .then(({body: {message}}) => {
+                expect(message).toBe('Bad Request');
+            })
+        })
+    })
```
Comment (Hard-Luck) [inline __tests__/app.test.js:283]: add a test that tests if the request body has additional keys and that they are ignored

Comment (Hard-Luck) [reply __tests__/app.test.js:283]: Needs a test for if the id in the url is not a number

File: db/controllers/controllers.js [hunk 3]
```diff
@@ -10,6 +14,44 @@ const getTopics = (request, response, next) => {
...
+        response.status(200).send({comments})
+    })
+    .catch(err => next(err))
+}
+
+const postCommentByArticleId = (request, response, next) => {
+    const {article_id} = request.params;
+    const newCommentData = request.body;
+
+    if (Object.keys(newCommentData).length <= 1 ||
```
Comment (Hard-Luck) [inline db/controllers/controllers.js]: This should be handled in the model.

**Review by Hard-Luck — CHANGES_REQUESTED** [review]

File: db/models/models.js [hunk 4]
```diff
@@ -10,6 +10,88 @@ const fetchTopics = () => {
...
+    const {username, body, votes = 0} = newCommentData
+
+    // checking if the username exists in the users database
+    const userString =  
+    `
+        SELECT * FROM users
+        WHERE username = $1;
+    
+    `
+    return db.query(userString, [username])
```
Comment (itsaliabdallah) [inline db/models/models.js:61]: this query wouldn't be needed, if you tried to add a user to the databse that didn't exist PSQL would throw an error

Comment (aloush-dev) [inline db/models/models.js:61]: this query wouldn't be needed, if you tried to add a user to the databse that didn't exist PSQL would throw an error

**Review by aloush-dev — COMMENTED** [review]
Really nice testing.
Just check your models where I left comments you're doing a query which wouldn't need to be done.

**Pull request merged**